core/src/main/scala/io/github/timwspence/cats/stm/STMLike.scala

```diff
@@ -63,7 +65,8 @@ trait STMLike[F[_]] {
   }
 
   object TVar {
-    def of[A](a: A)(implicit F: Concurrent[F]): Txn[TVar[A]] = Alloc(F.ref(a))
+    def of[A](a: A): Txn[TVar[A]] =
```

Comment:
Do we need this change? I'd rather not expose `newTVar` in the public API

Reply:
I think we do need something _like_ this. Calling `F.ref` obviously needs (at least) a `GenConcurrent`. If we want to remove that constraint, we need "someone else" to either (1) create the `Ref` (or `TVar` directly), or (2) provide the constraint. That someone else here is the implementation of the `STMLike` trait. The alternative (2) would be something like a `def concurrent: Concurrent[F]` in `STMLike`. I thought the `newTVar` is better, because it gives more options to the implementation (it can decide _how_ to create a tvar).

Reply:
🤦‍♂️ This is why I shouldn't review code when sleep-deprived from a new baby 😂 You're absolutely right. I think I actually prefer the Scato encoding (`def concurrent: Concurrent[F]` as you said) - we're not really expecting alternative implementations here and it makes the API more uniform as there's one way to instantiate a `TVar`.

We should be careful not to make it implicit though as standard usage is to `import stm._` and that might lead to ambiguous implicits for users